### PR TITLE
fix: completions after ```=``` in ```def ...```

### PR DESCRIPTION
**Pull request by Daniel-Anker-Hermansen**
related to https://github.com/flix/flix/issues/4554#issuecomment-1344833498

### CODE REVIEW

**Comment by Daniel-Anker-Hermansen** [comment]
@magnus-madsen I am only able to reproduce this if i make sure to not have the project in a compileable state at any time from launching the lsp server to asking for the suggestion, which is as expected since the root is null.

**Comment by magnus-madsen** [comment]
Let me try again then :) and report back

**Comment by magnus-madsen** [comment]
Still does not work. 

If I write `def main(): Unit \ IO = Lis` I get no completions (even if the program compiled correctly).

**Comment by Daniel-Anker-Hermansen** [comment]
Weird. This is how it looks for me:
image

**Comment by JonathanStarup** [comment]
On nightly 13dec I see the same as Daniel. The autocomplete is pretty good 👍

**Comment by magnus-madsen** [comment]
> 

That's probably an artifact of how the inferred type is checked (but not unified!) with the declared type.

**Comment by magnus-madsen** [comment]
> > > Random thing: I've found that x? autocomplete doesn't really know the result type if its in tail position
> > 
> > 
> > That's probably an artifact of how the inferred type is checked (but not unified!) with the declared type.
> 
> Huh, i wouldn't have guessed. What would the issue be with unifying them?

It has to be one sided. Consider:

```
def f(): List[Int32] = Nil // Yep, OK
def f(): List[a] = 1 :: Nil // Ooops, not OK
```

Makes sense?

**Comment by JonathanStarup** [comment]
Ah yeah okay, but there would be no downside of the body remembering the instantiation right?

**Comment by mlutze** [comment]
> > > > Random thing: I've found that x? autocomplete doesn't really know the result type if its in tail position
> > > 
> > > 
> > > That's probably an artifact of how the inferred type is checked (but not unified!) with the declared type.
> > 
> > 
> > Huh, i wouldn't have guessed. What would the issue be with unifying them?
> 
> It has to be one sided. Consider:
> 
> ```
> def f(): List[Int32] = Nil // Yep, OK
> def f(): List[a] = 1 :: Nil // Ooops, not OK
> ```
> 
> Makes sense?

If `a` is rigid then we get the appropriate error.

**Comment by magnus-madsen** [comment]
Appears fixed.
